### ISSUE TITLE
hxml comment character - doesn't do what it says

### ISSUE THREAD

**Issue by jcward** [HaxeFoundation/haxe#9505]
Tested in Haxe 4.1.1 on Linux 64-bit

**Issue:**
The [hxml spec page](https://haxe.org/manual/compiler-usage-hxml.html) of the Haxe manual states (emphasis mine):

> Inside .hxml files use a hash (i.e. #) to comment out **the rest of the line.**

This idea is consistent with, e.g., shell scripts, where you can write comments on the end of the line:

```shell
date +%s > .timestamp # Write the unix timestamp to a file
```

However, this doesn't appear to work for .hxml files. If I put a comment on a `-cp`, it silently fails to include the path:

```hxml
-cp /path/to/someLib # include the library
```

Results in: `type not found : pkg.SomeType`

If I put a comment on the `-main` line, it errors:

`Error: Could not process argument Main # Some comment`

**Solutions:**

**Preferred:** If it's not difficult, it would be nice if the `.hxml` parser supported the "rest of line comment" as described.

**Alternate:** If changing the hxml parser is difficult, you could instead update the manual page ([07-compiler-usage.md](https://github.com/HaxeFoundation/HaxeManual/blob/master/content/07-compiler-usage.md)) to say something like:

> Inside .hxml files, lines starting with a hash (i.e. #) are ignored as comments

**Comment by RealyUniqueName** [comment]
I'm not sure if there could be any consequences. For example a hash character in a string literal for `--macro` argument. 
I've created a PR just in case, but I think the documentation should be fixed instead of aligning implementation to the documentation.
@Simn ?

**Comment by Simn** [comment]
I agree it's better to update the documentation. CLI arguments can get quite complicated and there may indeed be `#` characters in various places.
